tests/e2e/e2e_util_test.go

```diff
@@ -146,7 +168,17 @@ func (s *IntegrationTestSuite) submitProposal(c *chain.Chain) {
 
 	s.Require().Truef(
 		strings.Contains(outBuf.String(), "code: 0"),
-		"tx returned non code 0",
+		"tx returned non code 0; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
+	)
+
+	currentHeight := s.chainHeight(s.valResources[c.ChainMeta.Id][0].Container.ID)
+
+	s.Require().Eventually(
+		func() bool {
+			return s.chainHeight(s.valResources[c.ChainMeta.Id][0].Container.ID) > currentHeight+2
```

Comment:
What is happening here? Why are we getting `currentHeight` and then trying to eventually reach 2 above current?

Reply:
You need to hit a few blocks before the rpc server is started, otherwise the following test (ibc send) will fail. The three blocks here have passed every time and I have run it many times, so I feel safe with it hard coded. Lmk if you think of a better solution (I guess we could continue to ping the nodes rpc until it responds successfully, lmk if thats a better option)